[bug] conan install [WinError 5] Access is denied

### Describe the bug

Hi,

after upgrading from Conan 2.24.0 to 2.25.0 `conan install` sporadically fails on Windows 11 when downloading binaries from remotes (no compiler / specific profiles involved). This happens sporadically for different packages like boost.

Conan version 2.25.0
Windows 11 23H3
Python 3.13.3

```
boost/1.88.0: Package installed 365fcbb7767708b731c482dd639c8192eac77a98
boost/1.88.0: Downloaded package revision 212897b988074dc9d433faad28321d79
WARN: The os.replace() to put this item in the package storage has failed. Maybethere was a concurrent process that did it first.
    Item: boost/1.88.0#1cfd62a314dc609738f0b840a9181006:365fcbb7767708b731c482dd639c8192eac77a98#212897b988074dc9d433faad28321d79 package
    Msg: [WinError 5] Access is denied: 'C:\\Users\\ip138\\.conan2\\p\\d\\ea68d8d947ef8' -> 'C:\\Users\\ip138\\.conan2\\p\\boost5ee40e8f6678d'
ERROR: Traceback (most recent call last):
  File "C:\Python313\Lib\site-packages\conan\cli\cli.py", line 297, in main
    cli.run(args)
    ~~~~~~~^^^^^^
  File "C:\Python313\Lib\site-packages\conan\cli\cli.py", line 194, in run
    command.run(self._conan_api, args[0][1:])
    ~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python313\Lib\site-packages\conan\cli\command.py", line 197, in run
    info = self._method(conan_api, parser, *args)
  File "C:\Python313\Lib\site-packages\conan\cli\commands\install.py", line 48, in install
    deps_graph, lockfile, install_error = _run_install_command(conan_api, args, cwd)
                                          ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python313\Lib\site-packages\conan\cli\commands\install.py", line 96, in _run_install_command
    install_error = conan_api.install.install_binaries(deps_graph=deps_graph, remotes=remotes,
                                                       return_install_error=return_install_error)
  File "C:\Python313\Lib\site-packages\conan\api\subapi\install.py", line 50, in install_binaries
    installer.install(deps_graph, remotes, install_order=install_order)
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python313\Lib\site-packages\conan\internal\graph\installer.py", line 259, in install
    self._handle_package(recipe_layout, package, install_reference, handled_count,
    ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                         package_count)
                         ^^^^^^^^^^^^^^
  File "C:\Python313\Lib\site-packages\conan\internal\graph\installer.py", line 343, in _handle_package
    assert os.path.isdir(pkg_folder), "Pkg '%s' folder must exist: %s" % (str(pref), pkg_folder)
           ~~~~~~~~~~~~~^^^^^^^^^^^^
AssertionError: Pkg 'boost/1.88.0:365fcbb7767708b731c482dd639c8192eac77a98' folder must exist: C:\Users\ip138\.conan2\p\boost5ee40e8f6678d\p

ERROR: Pkg 'boost/1.88.0:365fcbb7767708b731c482dd639c8192eac77a98' folder must exist: C:\Users\ip138\.conan2\p\boost5ee40e8f6678d\p
```

The Process Monitor log shows that the WinError 5 originates from some downloaded files being scanned by AV software while `os.replace()` from https://github.com/conan-io/conan/pull/19510 is called (e.g. C:\Users\ip138\.conan2\p\d\ea68d8d947ef8\p\lib\libboost_charconv.lib). I am not sure if this is something Conan should care about (root cause is the AV software not Conan). However this looks very similar to https://github.com/conan-io/conan/issues/6560 where a workaround was implemented.

### How to reproduce it

Use `conan install` with downloading binaries from remotes, while another process like an AV software reads this downloaded files immediately.

--- a/conan/__init__.py
+++ b/conan/__init__.py
@@ -2,5 +2,5 @@ from conan.internal.model.conan_file import ConanFile
 from conan.internal.model.workspace import Workspace
 from conan.internal.model.version import Version
 
-__version__ = '2.25.0'
+__version__ = '2.26.0-dev'
 conan_version = Version(__version__)

--- a/conan/__init__.py
+++ b/conan/__init__.py
@@ -2,5 +2,5 @@ from conan.internal.model.conan_file import ConanFile
 from conan.internal.model.workspace import Workspace
 from conan.internal.model.version import Version
 
-__version__ = '2.26.0-dev'
+__version__ = '2.25.1'
 conan_version = Version(__version__)

--- a/conan/internal/util/files.py
+++ b/conan/internal/util/files.py
@@ -10,19 +10,9 @@ import time
 
 from contextlib import contextmanager
 
-from conan.api.output import ConanOutput
 from conan.errors import ConanException
 
 _DIRTY_FOLDER = ".dirty"
-
-
-def atomic_replace(src, dst, item):
-    try:
-        os.replace(src, dst)  # ATOMIC!!!
-    except OSError as e:
-        msg = ("The os.replace() to put this item in the package storage has failed. Maybe"
-               f"there was a concurrent process that did it first.\n    Item: {item}\n    Msg: {e}")
-        ConanOutput().warning(msg)
 
 
 def set_dirty(folder):
@@ -193,6 +183,23 @@ def _change_permissions(func, path, exc_info):
 
 
 if platform.system() == "Windows":
+    def atomic_replace(src, dst, item):
+        retries = 3
+        delay = 0.5
+        for i in range(retries):
+            try:
+                os.replace(src, dst)  # ATOMIC!!!
+                return
+            except OSError as e:
+                if i == retries - 1:
+                    msg = ("The os.replace() to put this item in the package storage has failed.\n"
+                           "If you have an antivirus, try to exclude the "
+                           "Conan cache from the antivirus software."
+                           f"    Item: {item}\n    Error: {e}")
+                    raise ConanException(msg)
+                time.sleep(delay)
+
+
     def rmdir(path):
         if not os.path.isdir(path):
             return
@@ -226,6 +233,16 @@ if platform.system() == "Windows":
                                          "Close any app using it and retry.")
                 time.sleep(delay)
 else:
+    def atomic_replace(src, dst, item):
+        try:
+            os.replace(src, dst)  # ATOMIC!!!
+        except OSError as e:
+            msg = ("The os.replace() to put this item in the package storage has failed. Maybe"
+                   f"there was a concurrent process that did it first.\n"
+                   f"    Item: {item}\n    Msg: {e}")
+            raise ConanException(msg)
+
+
     def rmdir(path):
         if not os.path.isdir(path):
             return

--- a/conan/internal/util/files.py
+++ b/conan/internal/util/files.py
@@ -10,6 +10,7 @@ import time
 
 from contextlib import contextmanager
 
+from conan.api.output import ConanOutput
 from conan.errors import ConanException
 
 _DIRTY_FOLDER = ".dirty"
@@ -191,12 +192,13 @@ if platform.system() == "Windows":
                 os.replace(src, dst)  # ATOMIC!!!
                 return
             except OSError as e:
+                msg = ("The os.replace() to put this item in the package storage has failed.\n"
+                       "If you have an antivirus, try to exclude the "
+                       "Conan cache from the antivirus software."
+                       f"    Item: {item}\n    Error: {e}")
                 if i == retries - 1:
-                    msg = ("The os.replace() to put this item in the package storage has failed.\n"
-                           "If you have an antivirus, try to exclude the "
-                           "Conan cache from the antivirus software."
-                           f"    Item: {item}\n    Error: {e}")
                     raise ConanException(msg)
+                ConanOutput().warning(msg)
                 time.sleep(delay)
 
 

